pkg/provisioningv2/rke2/planner/registry.go

```diff
@@ -34,21 +34,20 @@ func toRegistryConfig(secrets v1.SecretCache, runtime, namespace string, registr
 
 	for registryName, config := range registry.Configs {
 		registryConfig := &registryConfig{}
-		if config.TLSSecretName == "" && config.InsecureSkipVerify {
+		if config.InsecureSkipVerify || config.TLSSecretName != "" || len(config.CABundle) > 0 {
 			registryConfig.TLS = &tlsConfig{
-				InsecureSkipVerify: true,
+				InsecureSkipVerify: config.InsecureSkipVerify,
 			}
-		} else if config.TLSSecretName != "" {
+		}
+
+		if config.TLSSecretName != "" {
```

Comment:
Since this used to be an else if, is there any reason why the secret stuff should not be configured if config.InsecureSkipVerify is true?

Reply:
I don't think so. In the original logic, if `config.TLSSecretName != ""`, then it didn't matter what `config.InsecureSkipVerify` was: it was get used on old line 49.